ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/validation/RequestValidator.java

```diff
@@ -166,7 +187,16 @@ private int validateRunScript(
         + "Note: RUN SCRIPT is deprecated and will be removed in the next major version. "
         + "statement: " + statement.getStatementText());
 
-    return validate(executionContext.parse(sql), statement.getOverrides(), sql);
+    // This is to maintain backwards compatibility until we deprecate
+    // RunScript in the next major release - the expected behavior was
+    // to return only the last entity
+    final List<ValidatedStatement> validatedStatements =
+        validate(executionContext, executionContext.parse(sql), statement.getOverrides(), sql);
+    for (int i = 0; i < validatedStatements.size() - 1; i++) {
```

Comment:
Rather than setting it to ignore resutls after the fact can we not make `Validatedstatement` immutable and set it at construction time?